#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 1 0 0 4.71238898038469
  position 0.5006122995624389 0.8385158370595016 -0.49896220065289193
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 1.5 1.5
}
Solid {
  translation 0.04 0.1 -0.02
  children [
    DEF cone Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
      }
      geometry Cone {
        bottomRadius 0.05
        height 0.1
      }
    }
  ]
}
DEF MY_ROBOT Robot {
  translation 0.5 0.05 -0.5
  children [
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF wheelLeftF HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "leftF"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation 3.9252239376397585e-17 7.850447875279517e-17 1 1.5708
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0.305882 0.898039 0.25098
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF wheelRightF HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "rightF"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation -1.96261196881988e-16 7.850447875279518e-17 1 1.5707999999999998
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF wheelLeftR HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "leftR"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation -1.1775671812919276e-16 3.9252239376397585e-17 1 1.5707999999999998
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF wheelRightR HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "rightR"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation 0 1.1775671812919276e-16 1 1.5707999999999995
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF disSensor DistanceSensor {
      translation 0 0 0.1
      rotation 0 1 0 -1.5707
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds"
    }
    DEF Cam Transform {
      translation 0 0.03 0.07
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.02 0.02 0.02
          }
        }
        Camera {
          translation 0 0.018 0
          rotation 0 1 0 3.1414
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Sphere {
                radius 0.015
                subdivision 3
              }
            }
          ]
          name "cam"
          fieldOfView 1
          width 640
          height 640
          far 10
          antiAliasing TRUE
        }
      ]
    }
  ]
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "supervisor"
  supervisor TRUE
}
